### PR TITLE
Add login alerts

### PR DESCRIPTION
**Pull request by codetheweb**
This creates a handy script during setup that sends an email alert with the IP and user whenever somebody logs onto your server.  Based on an Ask Ubuntu [answer](http://askubuntu.com/a/448602).

### CODE REVIEW

File: setup/login-alerts.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Create a script to be called when a user logs in
+cat << 'EOF' > /etc/ssh/login-alert.sh
+#!/bin/bash
```
Comment (bronson) [inline setup/login-alerts.sh]: These two comments are unnecessary.  (or maybe the "Create a script" should be reworded?)

**Comment by codetheweb** [comment]
A config option seems like a good idea.  Would it be appropriate to put it in `/etc/mailinabox.conf`?

**Comment by JoshData** [comment]
Hi.

This is an interesting idea but I don't think I will merge this without more discussion about what purpose this serves, and whether this is the right way to serve that purpose, etc etc.

**Comment by bronson** [comment]
Upsides:
- You can learn pretty quickly if something scary is afoot.

Downsides:
- You may learn to ignore the "you have logged in" emails.
- It only fires in response to PAM, so will miss remote code execution holes / root shells.
- More code to maintain, and PAM ain't exactly known for stability.

Since I don't log in much (yay miab stability!), I find the upside outweighs the downsides.

That said, this is a bit of a band aid approach.  If there's a better way to perform intrusion detection, then I'm all for that too.

**Comment by codetheweb** [comment]
@JoshData This aims to just be one more layer of security, like when iCloud or Google sends you an email saying someone logged into your account.  It doesn’t necessarily solve any security problem at the moment, it just keeps you informed about what’s happening.

A better way to increase security in general would be 2FA for both SSH and email login, although the first relies on 3rd party APIs and the second is impossible (for now).

If 2FA for SSH sounds interesting, it’s fairly easy to do with [Google](https://www.digitalocean.com/community/tutorials/how-to-protect-ssh-with-two-factor-authentication).



I agree that it's a bandaid solution.  I'm open to hear other ideas.

**Comment by yodax** [comment]
I don't think this will add much for security. If your system is breached by a a password or key leak, you are most likely not the person who responds in a timely manner to these messages any way. 

An installed root kit (most likely not installed via a ssh session) usually bypasses pam by using netcat and a custom dropbear binary. That wouldn't log any logins to the system. (Therefore it might be more interesting to expand the firewall checks to see if any additional ports are openend and that the default policy is deny) That would be a good warning sign for a new control port being openend that can bypass pam.

Adding this won't hurt for those points, however I also worry about pam being configured differently across different cloud providers.

FreeBSD sent daily status updates for security related issues. Bad su's, changed suid flags, added users, new services etc. It usually isn't the login to the system thats the indicator, it's all of these things together.

So in summary, I don't think it will hurt much, but I also think it will add very little in terms of security. I would filter these messages out.

**Comment by anoma** [comment]
I agree with @yodax that ` it will add very little in terms of security` but @bronson comment of `Since I don't log in much (yay miab stability!),` rings true here as well.

If I seen an email like this coming in day or night I would be acting on it immediately. That surely can only be a good thing even if it only helps 1 in 1000 users reduce the time before they take the box offline for repair.

**Comment by codetheweb** [comment]
I didn't think about cases where PAM could be bypassed @bronson and @yodax.  This was more designed with the scenario in mind that someone guesses your password or copies your key (first unlikely because of fail2ban, second slightly more likely if users aren't careful with their keys).  I've had a solution similar to this running on my box for ~1.5 years, and while I do get an alert at noon every day because of my backup script, it gives me peace of mind (even though it's probably an illusion).

At the end of the day, I'm not sure whether or not Mail-In-a-Box should set this up.  Having a config option during setup with the default 'No' might be best.

**Comment by codetheweb** [comment]
I'm closing this for now.  Those who want to add this to their server can do it without too much trouble.

File: setup/login-alerts.sh (hunk 1 above)
Comment (bronson) [inline setup/login-alerts.sh]: how strange.  it appears that the code was changed but my comment remains.

clearly there's nothing wrong with these two comments!
